Framework/Algorithms/src/FitPeaks.cpp

```diff
@@ -1333,10 +1353,17 @@ bool FitPeaks::processSinglePeakFitResult(
       if (left_bound < 0 || right_bound < 0)
         throw std::runtime_error("Code logic error such that left or right "
                                  "boundary of peak position is negative.");
-      if (peak_pos < left_bound || peak_pos > right_bound)
+      if (peak_pos < left_bound || peak_pos > right_bound) {
         peak_pos = -2.5;
-      else
+      } else if (peak_fwhm > (right_bound - left_bound)) {
+        // peak is too wide or window is too small
+        peak_pos = -2.75;
```

Comment:
And also -2.75? Shall we use an Enum instead?

Reply:
Yes an Enum is probably a good idea! It was just a negative (unphysical) number that indicates the reason the peak was excluded (as per the rest of that function). I suppose such information might be used for testing.